db/blip_handler.go

```diff
@@ -320,14 +320,22 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, opts *sendChangesOptions
 				for _, item := range change.Changes {
 					changeRow := bh.buildChangesRow(change, item["rev"])
 
-					if bh.purgeOnRemoval && len(change.Removed) > 0 && bh.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 {
-						userHasAccessToDoc, err := UserHasDocAccess(bh.db, change.ID, item["rev"])
-						if err != nil {
-							base.InfofCtx(bh.loggingCtx, base.KeySync, "Unable to obtain the doc: %s %s to verify user access: %v", base.UD(change.ID), item["rev"], err)
-							continue
-						}
-
-						if userHasAccessToDoc {
+					// If change is a removal and we're operating with protocol V3 and the change is not a tombstone fall
+					// into 3.0 removal handling
+					if change.allRemoved && bh.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 && !change.Deleted {
+						// If client requests revocations / removals via subChanges, continue work to possibly send
+						// removal. Otherwise skip sending removals.
```

Comment:
The nested continues here add code complexity.  In particular, it took me until now to realize that we're not sending a removal message when we get an error looking up access, which I think is incorrect?

What do you think about reorganizing along the lines of:
```
// If client doesn't want removals/revocations, don't send change
if !opts.revocations {
    continue
}

// If the user can access the removed doc through another channel, don't send change
userHasAccessToDoc, err == ...
if err == nil && userHasAccess {
    continue
}

// If we can't determine access due to error, log error and send change
if err != nil {
   log()
}

Reply:
Yeah thats fair, it was getting fairly nested.